packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/NamedTypeSymbolProvider.cs

```diff
@@ -306,19 +303,34 @@ private CSharpType ConstructCSharpTypeFromSymbol(
             string fullyQualifiedName,
             INamedTypeSymbol? namedTypeSymbol)
         {
+            var typeArg = namedTypeSymbol?.TypeArguments.FirstOrDefault();
             bool isValueType = typeSymbol.IsValueType;
             bool isEnum = typeSymbol.TypeKind == TypeKind.Enum;
-            var pieces = fullyQualifiedName.Split('.');
+            bool isNullable = typeSymbol.NullableAnnotation == NullableAnnotation.Annotated;
+            bool isNullableUnknownType = isNullable && typeArg?.TypeKind == TypeKind.Error;
+            string name = isNullableUnknownType ? fullyQualifiedName : typeSymbol.Name;
+            string[] pieces = fullyQualifiedName.Split('.');
+
+            // handle nullables
+            if (isNullable)
+            {
+                // System.Nullable`1[T] -> T
+                name = typeArg != null ? GetFullyQualifiedName(typeArg) : fullyQualifiedName;
```

Comment:
When would typeArg be null?

Reply:
If the typeSymbol can't be casted to a `INamedTypeSymbol` then `namedTypeSymbol` is null and there would be no type args. As far as which types that would be, I'm not entirely sure.